src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -55,11 +55,14 @@ public int getQuorumNumber() {
     @SuppressWarnings("unchecked")
     public void updateLayout(Layout layout, long rank)
             throws QuorumUnreachableException, OutrankedException, WrongEpochException {
+        runtime.invalidateLayout(); // Force our runtime to adopt latest epoch prior to updating
```

Comment:
This isn't a problem with the server-side or the protocol.  It's a client-side bug.  Using the counterexample:

    update_layout_qc:reset().
    update_layout_qc:update_layout(..., {layout,epoch=1,layouts=["sfritchie-m01:8000"],[],[]}, rank=1).
    update_layout_qc:update_layout(..., {layout,epoch=2,layouts=["sfritchie-m01:8000"],[],[]}, rank=21).

If the `invalidateLayout` call remains, both `updateLayout` calls are successful.

If I remove the `invalidateLayout` call, the second `updateLayout` call will fail because the client's CorfuRuntime (which is the same one used by both `updateLayout` calls) hasn't updated its epoch & layout view.  The server says:

     15:53:53.761 TRACE [event-4] o.c.i.LayoutServer - Incoming message with wrong epoch, got 0, expected 1, message was: LayoutRankMsg(super=LayoutMsg(super=CorfuMsg(clientID=152c7690-25f6-40fa-a356-625b1a6abd6f, requestID=2, epoch=0, buf=PooledSlicedByteBuf(freed), msgType=LAYOUT_PREPARE), layout=null), rank=1)

In examination of other client-side code, the way to tell the CorfuRuntime object to update to the correct view of the world appeared to be the `invalidateLayout()` method.

This PR is currently entangled with both layout-related changes and testing-related changes.  I wouldn't mind closing this PR and opening a new one when the dust settles.  Meanwhile, your question about using `invalidateLayout()` here is a good one. AFAICT, it needs to be done somewhere.  If this isn't the place, then that's a question to attach to the new PR.